tests/unit/src/test/scala/tests/ScalaToplevelSuite.scala

```diff
@@ -309,6 +382,52 @@ class ScalaToplevelSuite extends BaseSuite {
     dialect = dialects.Scala3,
   )
 
+  check(
+    "unapply",
+    """|package a
+       |
+       |object O {
+       | val (s1, s2) = ???
+       | var (s3, s4) = ???
+       | val Some(extr) = ???
+       | val CaseClass(extr1, extr2) = ???
+       | val (r, SomeConstructor) = ???
+       | val (p, 1) = (2, 1)
+       |}
+       |""".stripMargin,
+    List(
+      "a/", "a/O.", "a/O.s1.", "a/O.s2.", "a/O.s3().", "a/O.s4().", "a/O.extr.",
+      "a/O.extr1.", "a/O.extr2.", "a/O.r.", "a/O.p.",
+    ),
+    all = true,
+  )
+
+  check(
+    "additional-constructor",
+    """|package p
+       |
+       |class B() {
+       |  def this(i: Int) = ???
+       |}
+       |""".stripMargin,
+    List("p/", "p/B#"),
+    all = true,
+  )
+
+  check(
+    "case-class",
+    """|package p
+       |
+       |case class A(
+       |  name: Int, isH: Boolean = false
+       |) {
+       |  val V = ???
+       |}
+       |""".stripMargin,
+    List("p/", "p/A#", "p/A#name.", "p/A#isH.", "p/A#V."),
```

Comment:
We always did.

Reply:
Somehow I though that we wouldn't be indexing them since it's hard to say where the package is actually located (it exists in multple files.) Anyway, you can ignore this comment then.